File: 프로그래머스/0/120866. 안전지대/안전지대.java
class Solution {
    public int solution(int[][] board) {
        int answer = 0;
        
        int dx[] = {0,1,1,1,0,-1,-1,-1};
        int dy[] = {1,1,0,-1,-1,-1,0,1};
        
        int len = board.length;
        
        for(int i=0; i<len; i++){
            for(int j=0; j<len; j++){
                if(board[i][j] == 1){
                    for(int k=0; k<dx.length; k++){
                        if(i+dx[k] >= 0 && i+dx[k] < len && j+dy[k] >=0 && j+dy[k] < len) 
                            if(board[i+dx[k]][j+dy[k]] !=1 )board[i+dx[k]][j+dy[k]] =2;
                    }
                }
            }
        }
        
        for(int i =0; i<len; i++){
            for(int j=0; j<len; j++){
                if(board[i][j] == 0){
                    answer++;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120866. 안전지대/안전지대.py
def solution(board):
    answer = 0
    dx = [-1,0,1,1,1,0,-1,-1]
    dy = [1,1,1,0,-1,-1,-1,0]
    
    for x,row in enumerate(board):
        for y,point in enumerate(row):
            if point == 1 :
                for i in range(len(dx)):
                    if(x+dx[i] >=0 and x+dx[i] <len(board) and y+dy[i]>=0 and y+dy[i]<len(board) and board[x+dx[i]][y+dy[i]] == 0):
                        board[x+dx[i]][y+dy[i]] = 2
            
    for row in board:
        answer+= row.count(0)
    return answer